map-sdk/data-example/src/App.tsx

```diff
@@ -0,0 +1,194 @@
+import {FC, useCallback, useEffect, useRef, useState} from 'react';
+import {createMap, MapApi} from '@unfolded/map-sdk';
+import californiaCities from './data/california-cities.json';
+import arizonaCities from './data/arizona-cities.json';
+
+export const App: FC = () => {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const [map, setMap] = useState<MapApi | null>(null);
+
+  useEffect(() => {
+    const initMap = async () => {
+      const map = await createMap({
+        container: containerRef.current!
+      });
+
+      setMap(map);
+
+      // here you add map.thing()
+    };
+
+    initMap();
+  }, []);
+
+  return <div id="map-container" ref={containerRef}></div>;
+};
+
+export const AppWithCustomUI: FC = () => {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const [map, setMap] = useState<MapApi | null>(null);
+  const [datasetAdded, setDatasetAdded] = useState(false);
+  const [jsonPopupVisible, setJsonPopupVisible] = useState(false);
+  const [datasetData, setDatasetData] = useState<any>(null);
+
+  useEffect(() => {
+    const initMap = async () => {
+      const map = await createMap({
+        container: containerRef.current!
+      });
+
+      setMap(map);
+    };
+
+    initMap();
+  }, []);
+
+  // Function to get the dataset ID
+  // (in this example it is always the first/only dataset on the map)
+  const getDatasetId = (): string | null => {
+    if (!map) {
+      return null;
+    }
+
+    const datasets = map.getDatasets();
+
+    if (datasets.length === 1) {
+      return datasets[0].id;
+    } else {
+      console.log('There are no datasets or multiple datasets on the map.');
+      return null;
+    }
+  };
+
+  // Function to handle dataset update logic
+  const updateDataset = (datasetId: string, updateFunction: (dataset: any) => void) => {
```

Comment:
This can probably have a bit simpler signature, such as `const updateDataset = (datasetId: string, newData: Dataset) => void`, since we don't really use `updateFunction` for any other purpose, and we can leave the responsibility of providing the data to the caller.

Reply:
I went ahead and removed this entirely